src/app/parcels/batch/AddressEditCellInput.tsx

```diff
@@ -0,0 +1,219 @@
+import FreeFormTextInput from "@/components/DataInput/FreeFormTextInput";
+import { CenterComponent, GappedDiv } from "@/components/Form/formStyling";
+import { Button, Checkbox, FormControlLabel } from "@mui/material";
+import { useState } from "react";
+import { Address, BatchActionType, BatchTableDataState } from "@/app/parcels/batch/BatchTypes";
+import styled from "styled-components";
+import { ADDRESS_WIDTH } from "@/app/parcels/batch/ColumnWidths";
+import { postcodeRegex } from "@/app/clients/form/formSections/AddressCard";
+
+const GappedDivMargin = styled(GappedDiv)`
+    margin: 1rem;
+    width: calc(${ADDRESS_WIDTH}px - 2rem);
+    overflow: clip;
+`;
+
+const TopFreeFormTextInput = styled(FreeFormTextInput)`
+    margin-top: 0.5rem;
+`;
+
+interface AddressInputFields {
+    addressLine1?: string;
+    addressLine2?: string;
+    addressTown?: string;
+    addressCounty?: string;
+    addressPostcode?: string;
+}
+
+type AddressEditCellInputProps = {
+    tableState: BatchTableDataState;
+    id: number;
+    dispatchBatchTableAction: React.Dispatch<BatchActionType>;
+    simulateEscapeKeyPress: () => void;
+};
+
+const AddressEditCellInput: React.FC<AddressEditCellInputProps> = ({
+    tableState,
+    id,
+    dispatchBatchTableAction,
+    simulateEscapeKeyPress,
+}) => {
+    const batchRowAddress: Address | null = tableState.batchDataRows[id - 1].data.client.address;
+    const overrideRowAddress: Address | null = tableState.overrideDataRow.data.client.address;
+
+    const defaultAddressLine1: string | undefined =
+        id !== 0
+            ? batchRowAddress?.addressLine1 ?? undefined
+            : overrideRowAddress?.addressLine1 ?? undefined;
+    const defaultAddressLine2: string | undefined =
+        id !== 0
+            ? batchRowAddress?.addressLine2 ?? undefined
+            : overrideRowAddress?.addressLine2 ?? undefined;
+    const defaultAddressTown: string | undefined =
+        id !== 0
+            ? batchRowAddress?.addressTown ?? undefined
+            : overrideRowAddress?.addressTown ?? undefined;
+    const defaultAddressCounty: string | undefined =
+        id !== 0
+            ? batchRowAddress?.addressCounty ?? undefined
+            : overrideRowAddress?.addressCounty ?? undefined;
+    const defaultAddressPostcode: string | undefined =
+        id !== 0
+            ? batchRowAddress?.addressPostcode ?? undefined
+            : overrideRowAddress?.addressPostcode ?? undefined;
+    const [addressInputFields, setAddressInputFields] = useState<AddressInputFields>({
+        addressLine1: defaultAddressLine1,
+        addressLine2: defaultAddressLine2,
+        addressTown: defaultAddressTown,
+        addressCounty: defaultAddressCounty,
+        addressPostcode: defaultAddressPostcode,
+    });
```

Comment:
I think we can clean this up a bit.

1. Typescript is smart enough to work out types, so we don't need the `: string | undefined` declarations.
2. Some of the nullish operators `??` are redundant. Specifically, for address line 1, town and postcode, the variables are already constrained to `string | undefined` so we don't need them there. The others will need a check to convert `null` to `undefined`, but we can do this with one `??` not two by putting it outside the ternary operator. However, there may be a better way of doing it:
3. We can extract this into a helper method. Specifically, something that takes `(id, tableState) => AddressInputFields`. Something like this:
```
const getInitialAddressFields = (
    id: number,
    tableState: BatchTableDataState
): AddressInputFields => {
    const currentRowAddress: Address | null =
        id === 0
            ? tableState.overrideDataRow.data.client.address
            : tableState.batchDataRows[id - 1].data.client.address;
    return {
        addressLine1: currentRowAddress?.addressLine1,
        addressLine2: currentRowAddress?.addressLine2 ?? undefined,
        addressTown: currentRowAddress?.addressTown,
        addressCounty: currentRowAddress?.addressCounty ?? undefined,
        addressPostcode: currentRowAddress?.addressPostcode,
    };
};
```

Then the top of the component is just: 
```
const [addressInputFields, setAddressInputFields] = useState<AddressInputFields>(
    getInitialAddressFields(id, tableState)
);
```